packages/next/next-server/lib/router/router.ts

```diff
@@ -652,11 +651,16 @@ export default class Router implements BaseRouter {
   }
 
   /**
-   * Prefetch `page` code, you may wait for the data during `page` rendering.
+   * Prefetch page code, you may wait for the data during page rendering.
    * This feature only works in production!
-   * @param url of prefetched `page`
+   * @param url the href of prefetched page
+   * @param asPath the as path of the prefetched page
    */
-  prefetch(url: string): Promise<void> {
+  prefetch(
+    url: string,
+    asPath: string = url,
+    options: { priority?: boolean } = {}
```

Comment:
How about moving this type out of the arg so it can be re-used in `next/link` and then we can pass the options through directly since they will have the same type to avoid this extra destructuring in `next/link`'s `prefetch` 
```diff
+    var asPath = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : url;
+    var options = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};
```